plugins/modules/omevv_firmware_repository_profile.py

```diff
@@ -396,6 +425,98 @@ def execute(self):
             self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, diff={"before": {}, "after": {}}, changed=False)
 
 
+class ResyncFirmwareRepositoryProfile(FirmwareRepositoryProfile):
+
+    def __init__(self, module, rest_obj, rest_ome_obj):
+        self.module = module
+        super().__init__(module, rest_obj)
+        self.rest_obj = rest_obj
+        self.ome_obj = rest_ome_obj
+
+    def check_plugin_availability(self):
+        ump_plugin = 0
+        details = self.ome_obj.invoke_request("GET", "PluginService/Plugins")
+        resp = details.json_data["value"]
+        for item in resp:
+            if item.get("Name") == "Update Manager":
+                ump_plugin = 1
+                break
+        if not ump_plugin:
+            self.module.fail_json(msg=UMP_PLUGIN_NOT_FOUND_MSG, failed=True)
+
+    def check_mode_support(self):
+        omevv_profiles = self.omevv_profile_obj.get_firmware_repository_profile()
+        res = self.ome_obj.invoke_request("GET", "UpdateManagementService/Repositories")
+        ump_profiles = res.json_data["value"]
+        relevant_catalog_types = {"ESXi Catalog for Enterprise Servers", "vSAN Catalog for Enterprise Servers"}
+        filtered_ump_profiles = [p for p in ump_profiles if p["CatalogType"] in relevant_catalog_types]
+        not_avail = []
+        omevv_profile_names = {p["profileName"] for p in omevv_profiles}
+        for profile in filtered_ump_profiles:
+            if profile["Name"] not in omevv_profile_names:
+                not_avail.append(profile)
+        self.remove_keys(filtered_ump_profiles, not_avail)
+        if len(not_avail) > 0 and self.module._diff:
+            self.diff_dict['after'].update({"ump_profiles": not_avail})
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, diff=self.diff_dict, changed=True)
+        if len(not_avail) > 0:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        if len(not_avail) == 0 and self.module._diff:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, diff={"before": {}, "after": {}}, changed=False)
+        if len(not_avail) == 0:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, changed=False)
+
+    def remove_keys(self, filtered_ump_profiles, not_avail):
+        keys_to_remove = {"@odata.type", "@odata.id", "BaseCatalogID", "BaseCatalogName", "AvailableVersions", "DevicesRepo", "UrgentComponentsCount",
```

Comment:
why are we removing the keys, just curious ?

Reply:
Removing unnecessary keys which doesn't even gets reflected when profile is synced.